Mathlib/Order/RelSeries.lean

```diff
@@ -154,6 +157,148 @@ protected noncomputable def withLength [r.InfiniteDimensional] (n : ℕ) : RelSe
 lemma nonempty_of_infiniteDimensional [r.InfiniteDimensional] : Nonempty α :=
   ⟨RelSeries.withLength r 0 0⟩
 
+instance membership : Membership α (RelSeries r) :=
+  ⟨fun x s => x ∈ Set.range s⟩
+
+theorem mem_def {x : α} {s : RelSeries r} : x ∈ s ↔ x ∈ Set.range s :=
+  Iff.rfl
+
+/-- start of a series -/
+def bot (x : RelSeries r) : α := x 0
+
+/-- end of a series -/
+def top (x : RelSeries r) : α := x <| Fin.last _
```

Comment:
I think this deserves slightly longer docstrings. Also those names are inconsistent with `List.head`, `Fin.head`, etc...